cpp/src/core/radintrc.cpp

```diff
@@ -44,7 +44,6 @@ radTInteraction::radTInteraction()
 
 	NewMagnArray = NULL;
 	NewFieldArray = NULL;
-	IdentTransPtr = NULL;
```

Comment:
There is probably a ton of places in the code that may lead to some memory leak(s) when a method fails. In our previous / current interfaces (Mathematica. Python) this is not so important, because entire simulation is usually re-started if some method fails. But all such places could indeed de analyzed more thoroughly, and eliminated.
In this particular case, instead of a pointer to a radIdentTrans object, we can keep the object itself (since it does not require a lot of memory) as a member variable, in that or in the base class.
I am working now on parallelizing (via MPI) this and other functions, so will postpone this eventual change until the corresponding update.

Reply:
Yes, code with manual memory management tends to not do the right thing around errors, it's just very very hard to get it right. This is where things like `unique_ptr`, `vector` and other higher level constructs really shine. In fact, there is a wide consensus that since C++11 there is essentially never a need to use `new` or `delete` at all in application level code*, and it's considered best practice to avoid them.

Cool that you are working on parallelizing this code! Let me warmly recommend [ThreadSanitizer](https://clang.llvm.org/docs/ThreadSanitizer.html). It detects and pinpoints data races and other threading issues, which can be nearly impossible to fix or even find. If you write code that has any mutable state that is shared across threads you need this tool. It is one of the closest things to magic I have seen in the world of software.

*: The one exception to this is when creating `unique_ptr` objects, where you still need `new`. C++14 introduces [`std::make_unique`](https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique) to fix this little wart.